### PR TITLE
feat: Compatible with vcpkg package manager

### PR DESCRIPTION
**Pull request by denstiny**
vcpkg is the most commonly used package management tool for C++ users. Since neocmakelsp only searches the system's cmake package, it cannot adapt to the vcpkg package manager. This is regrettable, so I created this pr
It helps users search for packages installed by vcpkg

### CODE REVIEW

File: src/complete.rs [hunk 1]
```diff
@@ -6,7 +6,8 @@ use crate::fileapi;
 use crate::languageserver::BUFFERS_CACHE;
 use crate::scansubs::TREE_MAP;
 use crate::utils::treehelper::{get_pos_type, PositionType};
-use crate::{utils, CompletionResponse};
+use crate::utils::CACHE_CMAKE_PACKAGES_WITHKEYS;
+use crate::CompletionResponse;
```
Comment (Decodetalkers) [inline src/complete.rs:10]: this was a mistake , it should imported from tower_lsp::lsp_types, lol

File: src/utils/findpackage/vcpkg.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    tojump,
+                    from: "Vcpkg".to_string(),
+                });
+        }
+    }
+    packages
+}
+
+pub fn make_vcpkg_package_search_path(search_path: &Path) -> std::io::Result<Vec<String>> {
+    let lib_paths: Vec<&'static str> = [
```
Comment (Decodetalkers) [inline src/utils/findpackage/vcpkg.rs:138]: emm. I do not think 'static is needed

File: src/utils/findpackage/vcpkg.rs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        "x64-linux",
+        "x86-linux",
+        "x64-windows",
+        "x86-windows",
+        "x64-osx",
+    ]
+    .to_vec();
```
Comment (Decodetalkers) [inline src/utils/findpackage/vcpkg.rs:145]: this is also not needed I think

File: src/utils/findpackage/vcpkg.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    version,
+                    tojump,
+                    from: "Vcpkg".to_string(),
+                });
+        }
+    }
+    packages
+}
+
+pub fn make_vcpkg_package_search_path(search_path: &Path) -> std::io::Result<Vec<String>> {
```
Comment (Decodetalkers) [inline src/utils/findpackage/vcpkg.rs:137]: there is no error here, I think there is no need to return Result

**Comment by Decodetalkers** [comment]
there is some clippy worning. Emm you can run the clippy check

File: src/utils/findpackage/vcpkg.rs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+pub static VCPKG_PREFIX: LazyLock<Arc<Mutex<Vec<&str>>>> =
+    LazyLock::new(|| Arc::new(Mutex::new([].to_vec())));
+
+pub static VCPKG_LIBS: LazyLock<Arc<Mutex<Vec<&str>>>> =
+    LazyLock::new(|| Arc::new(Mutex::new([].to_vec())));
+
+fn get_available_libs() -> Vec<PathBuf> {
+    let mut ava: Vec<PathBuf> = vec![];
+    for prefix in VCPKG_PREFIX.lock().unwrap().iter() {
```
Comment (Decodetalkers) [inline src/utils/findpackage/vcpkg.rs:29]: This is bad.. you should

```rust
let vcpkg_prefix = VCPKG_PREFIX.lock().unwrap();
let vcpkg_libs = VCPKG_LIBS.lock().unwrap()
```

then use the locked variable

**Review by Decodetalkers — CHANGES_REQUESTED** [review]

**Comment by Decodetalkers** [comment]
Thanks for your effort!

**Comment by meinzer1899** [comment]
I did not understand what this PR adds. 
Can you help me?

**Comment by Decodetalkers** [comment]
> I did not understand what this PR adds. Can you help me?

when use you are using vcpkg, it will try to find packages under vcpkg folder, if you are not using vcpkg, it will behave like before. Just find package from additional place

**Pull request merged**